RiotSharpTest/RateLimiterTest.cs

```diff
@@ -0,0 +1,312 @@
+            var tasks = Enumerable.Range(Limit * 2, Limit).Select(i => Task.Delay(expectedDelayed)
+                .ContinueWith(task => RateLimiter.HandleRateLimitAsync())
+                .ContinueWith(task =>
+                {
+                    AssertDelayed(expectedDelayed, i);
+                })).ToList();
+            // immediate tasks`
+            var expected = TimeSpan.Zero;
+            tasks.AddRange(Enumerable.Range(0, Limit).Select(i => RateLimiter.HandleRateLimitAsync()
+                .ContinueWith(task =>
+                {
+                    AssertDelayed(expected, i);
+                })));
+
+            await Task.WhenAll(tasks);
+            foreach (var task in tasks)
+            {
+                Assert.IsNull(task.Exception);
+            }
+        }
+
+        /// <summary>
+        /// Sends 10 requests, waits 20 seconds, sends another 10 requests. Expects requests to unblock immediately.
+        /// </summary>
+        [TestMethod]
+        [TestCategory("RateLimiter")]
+        [Timeout(1000 * 10 * 3)]
+        public void AlternatingRequests()
+        {
+            for (var i = 0; i < Limit; i++)
+            {
+                RateLimiter.HandleRateLimit();
+                AssertDelayed(TimeSpan.Zero);
+            }
+            var delay = TimeSpan.FromTicks(TenSeconds.Ticks * 2);
+            Task.Delay(delay).Wait();
+            for (var i = 0; i < Limit; i++)
+            {
+                RateLimiter.HandleRateLimit();
+                AssertDelayed(delay);
+            }
+        }
+
+        /// <summary>
+        /// Same as above, but async.
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod]
+        [TestCategory("RateLimiter"), TestCategory("Async")]
+        [Timeout(1000 * 10 * 3)]
+        public async Task AlternatingRequestsAsync()
+        {
+            var expectedDelayed = TimeSpan.FromTicks(TenSeconds.Ticks * 2);
+            // delayed tasks first, just for fun
+            var tasks = Enumerable.Range(Limit*2, Limit).Select(i => Task.Delay(expectedDelayed)
+                .ContinueWith(task => RateLimiter.HandleRateLimitAsync())
+                .ContinueWith(task =>
+                {
+                    AssertDelayed(expectedDelayed, i);
+                })).ToList();
+            // immediate tasks
+            var expected = TimeSpan.Zero;
+            tasks.AddRange(Enumerable.Range(0, Limit).Select(i => RateLimiter.HandleRateLimitAsync()
+                .ContinueWith(task =>
+                {
+                    AssertDelayed(expected, i);
+                })));
+
+            await Task.WhenAll(tasks);
+            foreach (var task in tasks)
+            {
+                Assert.IsNull(task.Exception);
+            }
+        }
+
+        /// <summary>
+        /// Sends 5 requests. Waits 5 seconds. Sends 10 More requests. Expects first 10 requests to not block, rest to block by 10 seconds.
```

Comment:
Can you format comment so that they don't exceed 100 chars? It's the guideline we've followed so far.